client.go

```diff
@@ -32,6 +32,8 @@ var (
 	ErrReconnectFailed = errors.New("reconnect failed")
 	// ErrDuplicateSubscription ...
 	ErrDuplicateSubscription = errors.New("duplicate subscription")
+	//ErrConnectionClosed
```

Comment:
can we reuse existing `ErrClientClosed` for functionality introduced in this pr? At first glance we don't need new error here.